rcode-reader/src/main/java/eu/rssw/pct/elements/QueryElement.java

```diff
@@ -8,29 +8,37 @@
 import eu.rssw.pct.RCodeInfo;
 
 public class QueryElement extends AbstractAccessibleElement {
-  protected int prvte;
-  protected int cacheSize;
-  protected int flags;
-  protected int crc;
-  private String[] bufferNames;
+  private final String[] bufferNames;
+  private final int prvte;
+  private final int flags;
+  // protected int cacheSize;
+  // protected int crc;
 
-  public QueryElement(String name, Set<AccessType> accessType, byte[] segment, int currentPos, int textAreaOffset, ByteOrder order) {
+  public QueryElement(String name, Set<AccessType> accessType, String[] buffers, int flags, int prvte) {
     super(name, accessType);
+    this.bufferNames = buffers;
+    this.flags = flags;
+    this.prvte = prvte;
+  }
 
+  public static QueryElement fromDebugSegment(String name, Set<AccessType> accessType, byte[] segment, int currentPos,
+      int textAreaOffset, ByteOrder order) {
     int bufferCount = ByteBuffer.wrap(segment, currentPos, Short.BYTES).order(order).getShort();
-    this.prvte = ByteBuffer.wrap(segment, currentPos + 2, Short.BYTES).order(order).getShort();
-    this.cacheSize = ByteBuffer.wrap(segment, currentPos + 4, Short.BYTES).order(order).getShort();
-    this.flags = ByteBuffer.wrap(segment, currentPos + 6, Short.BYTES).order(order).getShort();
-    this.crc = ByteBuffer.wrap(segment, currentPos + 8, Short.BYTES).order(order).getShort();
+    int prvte = ByteBuffer.wrap(segment, currentPos + 2, Short.BYTES).order(order).getShort();
+    int flags = ByteBuffer.wrap(segment, currentPos + 6, Short.BYTES).order(order).getShort();
+    // int cacheSize = ByteBuffer.wrap(segment, currentPos + 4, Short.BYTES).order(order).getShort();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3ACommentedOutCodeLine)